src/commands/validate_enum.yml

```
@@ -0,0 +1,19 @@
+description: Validate parameter is an allowed value
+parameters:
+  key:
+    description: The name of the parameter to validate
+    type: string
+  value:
+    description: The value of the parameter to validate
+    type: string
+  enum:
+    description: The space-delimited list of accepted values
+    type: string
+steps:
+  - run:
+      name: Verify << parameters.key >> is an allowed value
+      command: |
+        if ! echo "<< parameters.enum >>" | grep -Eq "(^|[[:space:]])"<< parameters.value >>"($|[[:space:]])" ; then
```

What I think this is doing is checking if the string `parameter.value` appears at the start or the end of `parameters.enum` or is space separated from the other values, right?

You could consider:
```suggestion
        if ! echo "<< parameters.enum >>" | grep -Eq "(^|\s)<< parameters.value >>($|\s)" ; then
```

Depending on what's a legit value for the enum, you could consider using `\b` instead of `(^|\s)` / `($|\s)`